src/NATS.Client.JetStream/Models/ConsumerPauseResponse.cs

```diff
@@ -0,0 +1,22 @@
+using NATS.Client.JetStream.Internal;
+
+namespace NATS.Client.JetStream.Models;
+
+/// <summary>
+/// A response from the JetStream $JS.API.CONSUMER.PAUSE API
+/// </summary>
+public record ConsumerPauseResponse
+{
+    [System.Text.Json.Serialization.JsonPropertyName("paused")]
+    [System.Text.Json.Serialization.JsonIgnore(Condition = System.Text.Json.Serialization.JsonIgnoreCondition.Never)]
+    public bool Paused { get; set; }
```

Comment:
As this is a server response, I'd imagine this should be exposed as an immutable type ( see https://learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/immutability ) -- in short, the properties should be `{ get; init; }` to allow the JSON serializer to function correctly while preventing the caller from changing the properties (or more to the point, believing that changing them may have some kind of effect.) With this mind, the `Paused` property should be named `IsPaused` to align better with convention

Reply:
(The same should apply for other immutable return values here)

Reply:
should we make this change in others? (not in this PR of course)

Reply:
Have renamed to `IsPaused`, as well as using the `{ get; init; }` in the `ConsumerPauseResponse`.

Think it makes sense to have other server responses (as well as `ConsumerInfo` for example) use the same `{ get; init; }`.
But indeed like @mtmk says, would agree that would be good for another PR :slightly_smiling_face:

Reply:
@oising, it looks like `init` isn't supported in generated serializers in net6:
> System.InvalidOperationException : Deserialization of init-only properties is currently not supported in source generation mode.
we either ifdef or wait until net6 eol

Reply:
Ah, I forgot about net6 :/

> should we make this change in others? (not in this PR of course)

Ideally, yeah. It may technically be a breaking change if people are erroneously using the setters, but arguably their code is broken anyway.

Reply:
I'm curious about something -- following the thread here ( https://github.com/dotnet/runtime/issues/58770 ), that leads to the net7 PR here ( https://github.com/dotnet/runtime/pull/68064 ) -- switching to a positional syntax for the record, i.e. `public record Foo (bool Foo, string Bar);` also emits init only properties, but the source generated code for a net6 target will work fine, as long as you compile with an SDK greater than 6 (i.e. sdk 7 or 8 .) It seems that full support for init/required properties arrived in https://github.com/dotnet/runtime/pull/79828 and this only works for net8 targets. So, switching to position syntax for the record may work for net6 targets, if we compile with SDK8

Reply:
Ah, I remember now that "for reasons" you guys cannot down-target net6 using the net8 SDK because of how it pulls in dependencies and they will clash with other things.  It actually works fine for net6 targeting with SDK8, using both syntaxes (init-only or positional), and works only with positional with SDK7. For SDK6, it's broken as you state.

Reply:
I tested with the SDK8 before I remembered the limitation:

```csharp
// See https://aka.ms/new-console-template for more information

// Define the record type Foo

using System.Text.Json;
using System.Text.Json.Serialization;

// Create an instance of Foo
FooPos foo1 = new FooPos(true, "Test");

FooProp foo2 = new FooProp()
{
    BoolValue = true,
    StringValue = "Test"
};

// Create an instance of the context
var context = new MyJsonContext();

// Serialize the Foo instance to JSON using the context
string json = JsonSerializer.Serialize(foo1, context.FooPos);

Console.WriteLine($"Serialized FooPos: {json}");

var foo3 = JsonSerializer.Deserialize<FooPos>(json, context.FooPos);

Console.WriteLine($"Deserialized FooPos: {foo3}");


json = JsonSerializer.Serialize(foo2, context.FooProp);

Console.WriteLine($"Serialized FooProp: {json}");

var foo4 = JsonSerializer.Deserialize<FooProp>(json, context.FooProp);

Console.WriteLine($"Deserialized FooProp: {foo4}");
public record FooPos(bool BoolValue, string StringValue);

public record FooProp
{
    public bool BoolValue { get; init; }
    public string? StringValue { get; init; }
}

// Define the context class for JSON serialization
[JsonSourceGenerationOptions(
    WriteIndented = true,
    GenerationMode = JsonSourceGenerationMode.Metadata)]
[JsonSerializable(typeof(FooPos))]
[JsonSerializable(typeof(FooProp))]
public partial class MyJsonContext : JsonSerializerContext
{
}
```
Works fine under SDK8 :/

![image](https://github.com/nats-io/nats.net.v2/assets/1844001/ceff0166-521d-40c1-81f3-ee1fc0fea480)

Reply:
So I guess just either ifdef, or defer the work until later - but it would be nice to start on the True Path now ;)

Reply:
nice work 💯 🚀 ifdef'ed it ;-) We should be good. Created an issue so we don't forget #435